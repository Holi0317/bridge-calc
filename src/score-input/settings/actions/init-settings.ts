import { createAction } from "typesafe-actions";
import { ActionTypes } from "../../../action-types";
import { GameState } from "../../reducer";

/**
 * Initiate state of settings section by given game state.
 * If the state is null or stage is end, settings state will return to default.
 * @param state - Game state of on-going session
 */
export const initSettingsAction = createAction(
  ActionTypes.INIT_SETTINGS,
  (state: GameState) => state
)();
