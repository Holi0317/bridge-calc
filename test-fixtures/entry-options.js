import { deepFreeze } from "./deep-freeze";

export const defaultState = deepFreeze({
  rounds: 13,
  startingRound: 1,
  playerNames: ["John", "Mary", "Henry", "Joe"],
  importOpened: false
});

/**
 * Create mock redux state tree for test related to entry module.
 *
 * @param {object} rest Object to override default state in return tree
 * @returns {object} partial redux tree for testing
 */
export function makeEntryTree(rest) {
  return {
    entry: {
      ...defaultState,
      ...rest
    }
  };
}
