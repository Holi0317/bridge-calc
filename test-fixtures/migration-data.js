export const round1 = {
  'Bridge.currentRound': '1',
  'Bridge.maker': '0',
  'Bridge.players': '[{"bid":null,"win":null,"_buffer":{"lastRound":null,"combo":null,"miss":null,"maxCombo":null,"sum":null},"score":[],"name":"Jonathan"},{"bid":null,"win":null,"_buffer":{"lastRound":null,"combo":null,"miss":null,"maxCombo":null,"sum":null},"score":[],"name":"Helen"},{"bid":null,"win":null,"_buffer":{"lastRound":null,"combo":null,"miss":null,"maxCombo":null,"sum":null},"score":[],"name":"Humphrey"},{"bid":null,"win":null,"_buffer":{"lastRound":null,"combo":null,"miss":null,"maxCombo":null,"sum":null},"score":[],"name":"Margery"}]',
  'Bridge.state': '1',
  'Bridge.totalRounds': '13'
}

export const round2 = {
  'Bridge.currentRound':'2',
  'Bridge.maker':'1',
  'Bridge.players':'[{"bid":null,"win":null,"_buffer":{"lastRound":1,"combo":1,"miss":0,"maxCombo":0,"sum":11},"score":[11],"name":"Jonathan"},{"bid":null,"win":null,"_buffer":{"lastRound":1,"combo":0,"miss":1,"maxCombo":0,"sum":-1},"score":[-1],"name":"Helen"},{"bid":null,"win":null,"_buffer":{"lastRound":1,"combo":0,"miss":1,"maxCombo":0,"sum":-1},"score":[-1],"name":"Humphrey"},{"bid":null,"win":null,"_buffer":{"lastRound":1,"combo":0,"miss":1,"maxCombo":0,"sum":-1},"score":[-1],"name":"Margery"}]',
  'Bridge.state':'1',
  'Bridge.totalRounds':'13'
}
