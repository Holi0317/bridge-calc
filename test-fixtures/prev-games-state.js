export function makePrevGamesTree(prevGames) {
  return {
    prevGames: {
      prevGames
    }
  };
}
