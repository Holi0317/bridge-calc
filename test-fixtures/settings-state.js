import {deepFreeze} from './deep-freeze'

export const defaultState = deepFreeze({
  maker: null,
  names: {}
})
