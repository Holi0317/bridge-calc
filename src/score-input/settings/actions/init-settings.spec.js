import {initSettingsAction} from './init-settings'
import {waitingBidState} from '../../../../test-fixtures/current-game-states'

test('it should return init settings action', () => {
  const actual = initSettingsAction(waitingBidState)
  expect(actual).toMatchSnapshot()
})
