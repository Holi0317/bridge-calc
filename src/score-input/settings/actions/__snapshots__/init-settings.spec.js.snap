// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should return init settings action 1`] = `
Object {
  "payload": Object {
    "bid": Object {
      "a": 0,
      "b": 0,
      "c": 0,
      "d": 0,
    },
    "currentPlayerOrder": Array [
      "a",
      "b",
      "c",
      "d",
    ],
    "currentRound": 1,
    "id": "testing",
    "names": Object {
      "a": "John",
      "b": "Mary",
      "c": "Henry",
      "d": "Joe",
    },
    "rounds": 13,
    "scores": Object {
      "a": Array [],
      "b": Array [],
      "c": Array [],
      "d": Array [],
    },
    "stage": "waitingBid",
    "startTime": 0,
  },
  "type": "GAME_SETTINGS/INIT_SETTINGS",
}
`;
